addon/helpers/log-activity.ts

```diff
@@ -0,0 +1,39 @@
+import Helper from '@ember/component/helper';
+import { inject as service } from '@ember/service';
+import { assert } from '@ember/debug';
+import { getOwner } from '@ember/application';
+import ActivityTracking, { Activity } from '../services/activity-tracking';
+import { getOwnConfig } from '@embroider/macros';
+
+export default class extends Helper {
+  @service declare activityTracking: ActivityTracking;
+
+  compute(params: any[]) {
+    const [action, actionDescription] = params;
+    assert(
+      '[helper][log-activity] An action is required for this helper to work properly.',
+      typeof action === 'function'
+    );
+    assert(
+      '[helper][log-activity] An actionDescription needs to be passed for the activity-log to make sense.',
+      typeof actionDescription === 'string'
+    );
+
+    return (event: PointerEvent) => {
+      const activity = this.buildActivityObject(actionDescription);
+      this.activityTracking.log(activity);
+      return action(event);
```

Comment:
don't know if you tested but does it behave well if the action itself is more complex? like if we have have:`(log-activity (fn this.openModal "some_param") "Open the product modal")`

Reply:
Example of this exact usage is given in the documentation ^^ So yes ! it's been tested !